keps/sig-network/4410-k8s-network-interface/README.md

```diff
@@ -0,0 +1,102 @@
+# KEP-4410: Kubernetes Networking reImagined
+
+> **NOTE**: for the initial PR we've removed a lot of the templated text and
+> aimed to keep this first iteration small and easier to consume. We are only
+> focusing on the "What" and "Why" (e.g. motivation, goals, user stories) for
+> this iteration so that we can build consensus on those first before we add
+> any of the "How".
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+<!-- /toc -->
+
+## Summary
+
+This proposal is to design and implement the KNI [Kubernetes Networking Interface] or better known as Kubernetes Networking reImagined. KNI will create a Network resource and provide an API that will provide network status, availability, how to attach a pod to a network, detach the pod from the network and update a pods network.  
+
+## Motivation
+
+Kubernetes networking has traditionally been challenging to understand for users
+interacting with the Kubernetes API, and there has been considerable flexibility
+in how Container Network Interfaces (CNIs) set up networking within clusters.
+This has resulted in a scenario where things like pod networking (including pod
+to pod networking) is opaque to users, with different implementations taking
+markedly different approaches. This fragmentation has spread networking across
```

Comment:
So... KNI will reduce the flexibility in how pod networking can be set up, so that all implementations will have to take similar approaches?

Reply:
Reading this now and do see that this implies that we might be reducing flexibility which is the exact opposite of what we are trying to achieve.

Reply:
also the same with pod2pod networking is opaque to users aspect => Isn't that mostly how it should be as in end users only want the workloads to communicate correctly don't like 90% of end users not care about the "how" ? so opaque is not necessarily bad..right? :)

Reply:
and +1 to your previous comment:
```
reducing flexibility which is the exact opposite of what we are trying to achieve.
```
would love to see how we plan to achieve this, i.e bring a common API used by all but adoptable enough by different implementations..